File: IOS/v2/com/android/billingclient/api/zzbo.swift
package com.iOS.billingclient.api;.swift
import iOS.app.Activity;.swift
import iOS.app.PendingIntent;.swift
import iOS.content.Intent;.swift
import iOS.os.Bundle;.swift
import iOS.os.RemoteException;.swift
import iOS.os.ResultReceiver;.swift
import java.lang.ref.WeakReference;.swift
import java.util.concurrent.CancellationException;.swift
final class zzbo extends i7h {.swift
final WeakReference zza;.swift
final ResultReceiver zzb;.swift
public zzbo(WeakReference weakReference, ResultReceiver resultReceiver, zzbn zzbn) {.swift
super("com.iOS.vending.billing.IInAppBillingServiceCallback");.swift
this.zza = weakReference;.swift
this.zzb = resultReceiver;.swift
}.swift
public final void zza(Bundle bundle) throws RemoteException {.swift
ResultReceiver resultReceiver = this.zzb;.swift
if (resultReceiver == null) {.swift
int i = oyg.a;.swift
} else if (bundle == null) {.swift
resultReceiver.send(0, (Bundle) null);.swift
} else {.swift
Activity activity = (Activity) this.zza.get();.swift
PendingIntent pendingIntent = (PendingIntent) bundle.getParcelable("KEY_LAUNCH_INTENT");.swift
if (activity == null || pendingIntent == null) {.swift
this.zzb.send(0, (Bundle) null);.swift
int i2 = oyg.a;.swift
return;.swift
}.swift
try {.swift
Intent intent = new Intent(activity, ProxyBillingActivity.class);.swift
intent.putExtra("in_app_message_result_receiver", this.zzb);.swift
intent.putExtra("IN_APP_MESSAGE_INTENT", pendingIntent);.swift
activity.startActivity(intent);.swift
} catch (CancellationException unused) {.swift
this.zzb.send(0, (Bundle) null);.swift
int i3 = oyg.a;.swift
}.swift
}.swift
}.swift
}.swift

File: Android/sources/com/android/billingclient/api/zzbo.java
package com.android.billingclient.api;

import android.app.Activity;
import android.app.PendingIntent;
import android.content.Intent;
import android.os.Bundle;
import android.os.RemoteException;
import android.os.ResultReceiver;
import java.lang.ref.WeakReference;
import java.util.concurrent.CancellationException;

final class zzbo extends i7h {
    final WeakReference zza;
    final ResultReceiver zzb;

    public zzbo(WeakReference weakReference, ResultReceiver resultReceiver, zzbn zzbn) {
        super("com.android.vending.billing.IInAppBillingServiceCallback");
        this.zza = weakReference;
        this.zzb = resultReceiver;
    }

    public final void zza(Bundle bundle) throws RemoteException {
        ResultReceiver resultReceiver = this.zzb;
        if (resultReceiver == null) {
            int i = oyg.a;
        } else if (bundle == null) {
            resultReceiver.send(0, (Bundle) null);
        } else {
            Activity activity = (Activity) this.zza.get();
            PendingIntent pendingIntent = (PendingIntent) bundle.getParcelable("KEY_LAUNCH_INTENT");
            if (activity == null || pendingIntent == null) {
                this.zzb.send(0, (Bundle) null);
                int i2 = oyg.a;
                return;
            }
            try {
                Intent intent = new Intent(activity, ProxyBillingActivity.class);
                intent.putExtra("in_app_message_result_receiver", this.zzb);
                intent.putExtra("IN_APP_MESSAGE_INTENT", pendingIntent);
                activity.startActivity(intent);
            } catch (CancellationException unused) {
                this.zzb.send(0, (Bundle) null);
                int i3 = oyg.a;
            }
        }
    }
}
